examples/minimal_agent_production/steps/analyze.py

```diff
@@ -48,102 +44,125 @@ def perform_llm_analysis(
         content_preview = cleaned_content[:2000]
 
-        # Construct analysis prompt
-        prompt = f"""Analyze this document and provide structured analysis.
+        # Standard analysis request
+        analysis_request = """
+Respond with a JSON object containing:
+- summary: A concise 2-3 sentence summary
+- keywords: Array of 5 relevant keywords
+- sentiment: One of "positive", "negative", or "neutral"
+- readability: One of "easy", "medium", or "hard"
+"""
+
+        # Construct analysis prompt with JSON output format
+        prompt = f"""Analyze this document and provide structured analysis in JSON format.
 
 Document: {filename}
 Content: {content_preview}
 
-Provide a concise summary, relevant keywords, sentiment analysis, and readability assessment."""
+{analysis_request}
+
+Example:
+{{"summary": "This document discusses...", "keywords": ["ai", "machine", "learning", "data", "analysis"], "sentiment": "positive", "readability": "medium"}}"""
 
-        logger.debug(
-            f"Sending request to {model} with {len(prompt)} character prompt"
-        )
         start_time = time.time()
 
-        # Create instructor client with litellm
-        client = instructor.from_litellm(completion)
+        # Create OpenAI client
+        client = OpenAI()
 
-        # Call LLM service with structured output
+        # Call OpenAI API
         response = client.chat.completions.create(
             model=model,
-            response_model=AnalysisResponse,
             messages=[
                 {
                     "role": "system",
-                    "content": "You are a document analysis assistant. Provide structured analysis as requested.",
+                    "content": "You are a document analysis assistant. Always respond with valid JSON in the exact format requested.",
                 },
                 {
                     "role": "user",
                     "content": prompt,
                 },
             ],
             max_tokens=300,
-            metadata=metadata or {},
+            temperature=0.1,  # Low temperature for consistent output
         )
 
         end_time = time.time()
         latency_ms = int((end_time - start_time) * 1000)
 
-        # Extract structured response
-        analysis_response = response
+        # Parse JSON response
+        response_text = response.choices[0].message.content
+        if response_text is None:
+            response_text = ""
+        try:
+            analysis_response = json.loads(response_text)
+        except json.JSONDecodeError:
+            # Fallback if JSON parsing fails
+            analysis_response = {
+                "summary": "Document analysis completed",
+                "keywords": [
+                    "document",
+                    "analysis",
+                    "text",
+                    "content",
+                    "review",
+                ],
+                "sentiment": "neutral",
+                "readability": "medium",
```

Comment:
these sentiment and readability values should be an enum or a `Literal` type
defined somewhere else and just used here.